File: clojure/project-euler/src/problem1.clj
(ns problem1)

; Add all the natural numbers below one thousand that are multiples of 3 or 5.

(defn multiple-of-3-or-5? [x]
  (or
   (= (rem x 3) 0)
   (= (rem x 5) 0)))

(defn problem1 []
  (reduce +
          (filter multiple-of-3-or-5? (range 0 1000)))) 

File: haskell/project-euler/Problem1.hs
module Main where

isMultipleOf3Or5 :: Integer -> Bool
isMultipleOf3Or5 x
	| x `rem` 3 == 0 	= True
	| x `rem` 5 == 0 	= True
	| otherwise 		= False

problem1 = foldr (+) 0 (filter isMultipleOf3Or5 [0..999])

main = do
	print problem1
